scripts/rebar_lock_check

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+## Check rebar.lock is up-to-date.
+##
+## Copies `rebar.lock` file to temporary directory, then attempts to
+## re-generate it, checks it is the same, finally deletes the
+## temporary directory.
+##
+## The re-generation of the `rebar-lock` file is attempted by:
+## * First running `rebar3 unlock` - so to delete the old lock file;
+## * Then running `rebar3 upgrade` - so to generate a lock file
+##   attempting to minimize contamination by already-fetched
+##   dependencies (e.g. transitive ones) - if any.
+##
+## On failure neither the `rebar.lock` file nor the fetched
+## dependencies are restored.
+
+on_exit() {
+    rm -r "${TmpDir:?}"
+}
+
+set -e
+
+R="${1:?}" ## Absolute path of `rebar3` executable.
```

Comment:
Can you please use more descriptive variable names? A/B/C/D vars really does not make sense to me, even in bash, even with the great comments you added.

Some suggestions:
- `rebar_bin` or `RebarBin` 
- `working_bin`
- `rebar_lock_path`

(btw we should agree on some style guide and follow it, e.g. https://google.github.io/styleguide/shell.xml)

Reply:
In the past when I used more descriptive variable names I received comments to use short var names (one or two chars max)... I agree with some one style guide.